### PR TITLE
Added fixes for Julia 1.0

### PR DESCRIPTION
**Pull request by simonschoelly**
Most of the stuff works, but the package `Clp.jl` is not yet ready for version 1.0. Therefore the tests fail and `mincost_flow` does not work yet.

Should also remove most of the warnings in #18

### CODE REVIEW

**Comment by matbesancon** [comment]
oh wonderful thanks! 
For the parametric functions I'm not sure SimpleTraits.jl supports the `where` syntax

**Comment by matbesancon** [comment]
Also, I think we can change the travis config:
```
julia:
  - 0.7
  - 1.0
  - nightly

matrix:
  allow_failures:
- julia: nightly
```

**Comment by simonschoelly** [comment]
I changed the travis config. There is also appveyor config that defaults to version 0.6 but I'm not familiar with that. 
SimpleTraits does support the `where` syntax, you just have to be a bit careful when you have multiple arguments in the where clause.

**Comment by matbesancon** [comment]
To update appveyor if you can: https://github.com/JuliaCI/Appveyor.jl
The readme example is up to date for 1.0, copy-pasting should just work, uncomment the nightly fail allowed and we're god

**Comment by simonschoelly** [comment]
Appveyor is running now but Windows has problems with clp on all versions. The other tests all pass:
https://ci.appveyor.com/project/simonschoelly/lightgraphsflows-jl/build/1.0.10

**Comment by matbesancon** [comment]
yes clp is supposedly good on 0.7, not sure about 1.0 they didn't put it in their CI scripts, I'll PR them for it

**Comment by matbesancon** [comment]
https://github.com/JuliaOpt/Clp.jl/pull/32

**Comment by juliohm** [comment]
Thank you @simonschoelly for your work. What is the status?

**Comment by simonschoelly** [comment]
Until Clp.jl and Cbc.jl, which is a dependency for the former, work on 1.0, the tests can't pass.

See JuliaOpt/Clp.jl#31 and JuliaOpt/Cbc.jl#54.

**Comment by simonschoelly** [comment]
Clp runs on Julia v1.0 now, so this PR could probably be merged.

File: REQUIRE [hunk 1]
```diff
@@ -1,4 +1,4 @@
-julia 0.6
+julia 0.7
 LightGraphs 0.12
```
Comment (matbesancon) [inline REQUIRE]: should we up LightGraphs to v1.1?

**Review by matbesancon — CHANGES_REQUESTED** [review]

**Review by matbesancon — APPROVED** [review]

**Comment by matbesancon** [comment]
Wonderful, merging as soon as CI passes, thanks @simonschoelly !

**Pull request merged**